code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/identity/TokenResourceManager.java

```diff
@@ -0,0 +1,92 @@
+package io.cattle.platform.iaas.api.auth.identity;
+
+import io.cattle.platform.iaas.api.auth.SecurityConstants;
+import io.cattle.platform.iaas.api.auth.integration.github.GithubConstants;
+import io.cattle.platform.iaas.api.auth.integration.interfaces.TokenCreator;
+import io.cattle.platform.iaas.api.auth.TokenUtils;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.factory.SchemaFactory;
+import io.github.ibuildthecloud.gdapi.model.ListOptions;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.request.resource.impl.AbstractNoOpResourceManager;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+
+public class TokenResourceManager extends AbstractNoOpResourceManager {
+
+    private List<TokenCreator> tokenCreators;
+
+    public List<TokenCreator> getTokenCreators() {
+        return tokenCreators;
+    }
+
+    @Inject
+    public void setTokenCreators(List<TokenCreator> tokenCreators) {
+        this.tokenCreators = tokenCreators;
+    }
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[]{Token.class};
+    }
+
+    @Override
+    protected Object createInternal(String type, ApiRequest request) {
+        if (!StringUtils.equals(TokenUtils.TOKEN, request.getType())) {
+            return null;
+        }
+        try {
+            return createToken(request);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Token createToken(ApiRequest request) throws IOException {
+        Token token = null;
+        List<ClientVisibleException> exceptions = new ArrayList<>();
+        for (TokenCreator tokenCreator : tokenCreators) {
+            if (tokenCreator.isConfigured()){
+                try {
+                    token = tokenCreator.createToken(request);
+                } catch (ClientVisibleException e) {
+                    exceptions.add(e);
+                }
+                if (token != null) {
+                    break;
+                }
+            }
+        }
+        if (token == null && !exceptions.isEmpty()) {
+            for (Exception e : exceptions) {
+                e.printStackTrace();
+            }
+            if (exceptions.size() != 1) {
+                throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR,
+                        "AuthorizationProvider", "invalid code.",
+                        null);
+            } else {
+                throw exceptions.get(0);
+            }
+        } else if (token == null){
+            throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR,
```

Comment:
That would depend on who you ask. I think it is an error if you try to post to something that isn't configured. Its like trying to start a car with no gas vs broken batter broken batter is 500 no gas 404  this would be broken battery. At least IMO